### DISCUSSION TITLE
Compiler of Debugger-for-java

### DISCUSSION THREAD

**Discussion by dawosch** [microsoft/vscode-java-debug#1313] [General]
Hey, i want to know if this extension is using a specific java compiler?
When i build my java project with maven and my local installed java version in the source files of my classes is the compiler javac.

But when i start my project with the spring boot dashboard extension, the debugger extension always compiles my project and then the compiler changed to Eclipse JDT (IDE).

Maven compile step:
```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2023-03-18T11:50:30+0100",
    comments = "version: 1.5.3.Final, compiler: Eclipse JDT (IDE) 3.33.0.v20230218-1114, environment: Java 17.0.5 (Microsoft)"
)
```

Java Debugger compile step:
```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2023-03-18T11:52:09+0100",
    comments = "version: 1.5.3.Final, compiler: javac, environment: Java 17.0.5 (Microsoft)"
)
```

It seems that the eclipse compiler ignores my classpath, because the generated files are different.
With the javac compiler everything works, with the eclipse compiler values will not be settet.

The error occures only with the "MapStruct" dependency and a custom naming strategy.
[MapStruct installation](https://mapstruct.org/documentation/installation/)
[Custom naming strategy](https://mapstruct.org/documentation/stable/reference/html/#using-spi)

But maybe i'm on a wrong way. But this is the only different i noticed.

Edit:
Here i created a repo with all the files to reproduce the error: [Repository](https://github.com/dawosch/vscode-java-extension-compiler-error)

Step 1: Clone the repo
Step 2: run the command "mvn -U clean install" on the root folder
Step 3: Look at the file "TestMappterInterfaceImpl" in "app/target/generated-sources/annotations/con/example/demo"
You will notice that on line 20 the setter is generated
![image](https://user-images.githubusercontent.com/57100280/226125929-a638ad84-8932-4c53-afac-5b4552844209.png)

Step 4: Run the "app" project from the "Spring Boot Dashboard" extension
Step 5: Look at the same file
You will see that the compiler has changed and the setter isn't there
![image](https://user-images.githubusercontent.com/57100280/226126005-cc50571b-ae8a-4fdf-8b38-66b4f590ef52.png)

**Comment by testforstephen** [comment]
Java extension doesn't support delegating to 3rd party build tools for compilation yet, but this would be a direction we'd like to go forward in future.

Back to your case, the solution we need to figure out is how to make JDT compiler to support the MapStruct annotation better?
@jdneo any thoughts?

**Reply by jdneo** [reply]
Looks like the root cause is that JDT does not recognize the SPI config: https://github.com/dawosch/vscode-java-extension-compiler-error/blob/main/mapstruct-accessing-strategy/src/main/resources/META-INF/services/org.mapstruct.ap.spi.AccessorNamingStrategy.

Not sure if there is a workaround for this issue now.

**Comment by snjeza** [comment]
@jdneo @testforstephen See https://github.com/redhat-developer/vscode-java/issues/3007#issuecomment-1475466306

**Comment by kunwargupta** [comment]
![Screenshot (63)](https://user-images.githubusercontent.com/123477226/233823412-d743a8b8-f1cc-493c-aa1b-89eb2e77b0e1.png)
Failed to launch debuggee in the terminal. Reason: Failed to launch debuggee in terminal.
Reason: java.util.concurrent.TimeoutException: timeout ** Anyone please fix this error please help I'm stuck**
